include/sddf/serial/queue.h

```diff
@@ -167,27 +167,20 @@ static inline void serial_update_visible_head(serial_queue_handle_t *queue_handl
     queue_handle->queue->head = local_head;
 }
 
-static inline uint32_t serial_queue_contiguous_length(serial_queue_handle_t *queue_handle)
+static inline uint32_t serial_queue_length(serial_queue_handle_t *queue_handle)
 {
-    uint32_t head = queue_handle->queue->head;
-    uint32_t tail = queue_handle->queue->tail;
-
-    if (head <= tail) {
-        return tail - head;
-    } else {
-        return (UINT32_MAX - head) + 1;
-    }
+    return queue_handle->queue->tail - queue_handle->queue->head;
```

Comment:
This I think is correct in the context of what is calling `serial_queue_length` right now but this function is available to code outside of `queue.h` which means either this needs to also do modulo queue size or have a comment stating the invariant that tail >= head.

Reply:
Actually as they're unsigned the invariant is 0 <= (tail - head) <= SIZE